### DISCUSSION TITLE
Tutorials

### DISCUSSION THREAD

**Discussion by mwr2019** [codewars/codewars.com#2835] [Ideas]
### Suggestion for Tutorial into any langugage through Codewars.com

This is my suggestion for how to improve the experience on codewars for new coders or for anyone who wants to learn a new language.

Basically you have a button or an option to enter a language in "Tutorial mode" to let the user go through a series of steps to get introduced into the language. Instead of getting the normal kata with a problem, and  a required solution, there will be a normal kata with a problem, and a solution with comments and maybe a small task for the user to learn the key aspects of the language(variables, loops, arrays, functions etc.) so that the user will become somewhat accustomed to the syntax. This should also introduce the user to the language' documentation.
Lets say this is the first page of a 5-10 page quick tutorial for total beginners.
Authors could ofcourse cater to either beginners or advanced coders learning new languages so that based on the users preferences, a suitable tutorial will show up. The tutorials could also be provided by the community.

And so the reason why i think this should be a good addition is because when i started coding on codewars it was not so clear how i was going to find the answers to the katas. If i started a totally new language, i started googleing the problems etc. untill i realized, the best way is to find the simplest katas, hit unlock solutions, and read the lines untill i can understand it or look for a solution with good commenting. I believe the process of learning a new language for either an experienced coder or especially for a beginner coder, would be sped up with a feature like this.

Because of the great layout of Codewars, and the amout of data that could be used for the tutorials, i believe this would be a suitable addition to the site.

Here is my example:

**introduction to variables and values in Go**

This program does this:
prints my name and age. 
prints the number of fingers plus the number of toes that I have. 
prints true if the sum of my fingers and toes are equal to 10.

You can check out [https://gobyexample.com/](url) if you want to look at more examples or [https://pkg.go.dev/std](url) for documentation

Your job is to make your own print line at the end of the program that includes your name and and the summarization of your age, fingers and toes.

```
package main

// This is the importation of libraries
import "fmt"
import "strconv"

// Our main function
func main() {

    // Initialize variable a, and assign it the value "Mike"
    // a <-- "Mike"
    var a = "Mike"

    // Initialize variable b, and assign it the value 33 and similar for c and d
    // b <-- 33
    var b = 33

    // This is a type conversion. it also declares and intialized at the same time hence the :=. 
    // the function strconv.Itoa() converts an int to a string
    br := strconv.Itoa(b)
    var c = 5
    var d = 5

    // Initialize variable e, which is a boolean that can only store true or false. we set it to false since we dont 
    // know the answer yet.
    var e = false
    fmt.Println(e) // print for better output understanding

    // this is an if statement. it will summarise b and c and if that is the same as 10 it will give e the value true
    if b+c == 10 {
	e = true
    }
    fmt.Println("Hello! My name is "+a+" and i am " +br+ " years old.")
    fmt.Println(c+d, e)
    fmt.Println(e)

    // fmt.Println(yourprint)
}
```
The layout would just be all the commenting on the left, and the code editor on the right, as in the katas.

This is just an idea and my first post. Let me know what you think

**Comment by televisionia** [comment]
I think this is a really good idea, I could imagine the site gaining more attention with beginner-friendly tutorials. Tutorials with lots of interactivity can help you learn something a lot faster too. Just letting you know, the hyperlinks on your post don't work so I had to manually copy and paste the text.

**Comment by RubberizedMayhem** [comment]
That would be great because I find myself just pressing the unlock solutions button because I have no idea what is going on

**Reply by Blind4Basics** [reply]
Did you see this and the next pages?
https://docs.codewars.com/getting-started/finding-kata
